remotes/docker/resolver.go

```diff
@@ -401,9 +413,8 @@ func (r *dockerResolver) Resolve(ctx context.Context, ref string) (string, ocisp
 	}
 
 	// If above loop terminates without return, then there was an error.
-	// "firstErr" contains the first non-404 error. That is, "firstErr == nil"
-	// means that either no registries were given or each registry returned 404.
-
+	// "firstErr" contains the first error. That is, "firstErr == nil" means
+	// that either no registries were given.
```

Comment:
This comment could be updated further or just simplified to mention the below case handles no registries given

Reply:
Updated, thanks :pray: